Cannot use unique_ptr for unexpected value gcc5.5

This fails to compile with the following error that indicates that the deleted copy constructor is used.
```C++
tl::expected<int, std::unique_ptr<std::string>> func()
{
  return 1;
}
```

```
/mnt/c/w/repos/perf_tests/perf_tests/../sub/expected/include/tl/expected.hpp: In instantiation of ‘U& tl::expected<T, E>::value() & [with U = int; tl::detail::enable_if_t<(! std::is_void<_Res>::value)>* <anonymous> = 0u; T = int; E = std::unique_ptr<std::__cxx11::basic_string<char> >]’:
/mnt/c/w/repos/perf_tests/perf_tests/expected.cpp:161:37:   required from here
/mnt/c/w/repos/perf_tests/perf_tests/../sub/expected/include/tl/expected.hpp:1903:30: error: use of deleted function ‘std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = std::__cxx11::basic_string<char>; _Dp = std::default_delete<std::__cxx11::basic_string<char> >]’
       detail::throw_exception(bad_expected_access<E>(err().value()));
                              ^
In file included from /usr/include/c++/5/memory:81:0,
                 from /mnt/c/w/repos/perf_tests/perf_tests/expected.cpp:2:
/usr/include/c++/5/bits/unique_ptr.h:359:7: note: declared here
       unique_ptr(const unique_ptr&) = delete;
       ^
In file included from /mnt/c/w/repos/perf_tests/perf_tests/expected.cpp:5:0:
/mnt/c/w/repos/perf_tests/perf_tests/../sub/expected/include/tl/expected.hpp:1191:12: note:   initializing argument 1 of ‘tl::bad_expected_access<E>::bad_expected_access(E) [with E = std::unique_ptr<std::__cxx11::basic_string<char> >]’
   explicit bad_expected_access(E e) : m_val(std::move(e)) {}
```

Compiler:
g++ (Ubuntu 5.5.0-12ubuntu1~16.04) 5.5.0 20171010

--- a/include/tl/expected.hpp
+++ b/include/tl/expected.hpp
@@ -1907,14 +1907,14 @@ public:
             detail::enable_if_t<!std::is_void<U>::value> * = nullptr>
   TL_EXPECTED_11_CONSTEXPR const U &&value() const && {
     if (!has_value())
-      detail::throw_exception(bad_expected_access<E>(err().value()));
+      detail::throw_exception(bad_expected_access<E>(std::move(err()).value()));
     return std::move(val());
   }
   template <class U = T,
             detail::enable_if_t<!std::is_void<U>::value> * = nullptr>
   TL_EXPECTED_11_CONSTEXPR U &&value() && {
     if (!has_value())
-      detail::throw_exception(bad_expected_access<E>(err().value()));
+      detail::throw_exception(bad_expected_access<E>(std::move(err()).value()));
     return std::move(val());
   }
 

--- a/include/tl/expected.hpp
+++ b/include/tl/expected.hpp
@@ -18,7 +18,7 @@
 
 #define TL_EXPECTED_VERSION_MAJOR 1
 #define TL_EXPECTED_VERSION_MINOR 0
-#define TL_EXPECTED_VERSION_PATCH 0
+#define TL_EXPECTED_VERSION_PATCH 1
 
 #include <exception>
 #include <functional>

--- a/tests/issues.cpp
+++ b/tests/issues.cpp
@@ -127,3 +127,12 @@ TEST_CASE("Issue 49", "[issues.49]") {
     .and_then(test2);
 }
 #endif
+
+tl::expected<int, std::unique_ptr<std::string>> func()
+{
+  return 1;
+}
+
+TEST_CASE("Issue 61", "[issues.61]") {
+  REQUIRE(func().value() == 1);
+}

